core/conversion/converters/impl/expand.cpp

```diff
@@ -15,24 +16,93 @@ namespace converters {
 namespace impl {
 namespace {
 
+nvinfer1::ILayer* create_plugin(
+    ConversionCtx* ctx,
+    const torch::jit::Node* n,
+    nvinfer1::ITensor* inShape,
+    nvinfer1::ITensor* expandShape,
+    int32_t in_rank,
+    int32_t expand_rank,
+    const char* name) {
+  auto creator = new plugins::CheckShapePluginCreator();
+  std::vector<nvinfer1::PluginField> fields;
+  nvinfer1::PluginField input_rank("input_rank", &in_rank, nvinfer1::PluginFieldType::kINT32, 1);
+  nvinfer1::PluginField output_rank("expand_rank", &expand_rank, nvinfer1::PluginFieldType::kINT32, 1);
+  fields.push_back(input_rank);
+  fields.push_back(output_rank);
+  nvinfer1::PluginFieldCollection collection;
+  collection.nbFields = fields.size();
+  collection.fields = fields.data();
+  auto plugin = creator->createPlugin(name, &collection);
+
+  nvinfer1::ITensor* inputs[] = {inShape, expandShape};
+  auto expandShape_layer = ctx->net->addPluginV2(inputs, 2, *plugin);
+  TRTORCH_CHECK(expandShape_layer, "Unable to create interpolation plugin from node" << *n);
+
+  expandShape_layer->setName("CheckShapePlugin");
+  return expandShape_layer;
+}
+
+void addSliceInput(nvinfer1::Dims& dims, int idx, ConversionCtx* ctx, nvinfer1::ISliceLayer* slice) {
+  int32_t rank = static_cast<int32_t>(dims.nbDims);
+  int32_t* tmp = new int32_t[rank];
+  for (int i = 0; i < rank; i++)
+    tmp[i] = dims.d[i];
+  const nvinfer1::Dims d{1, {rank}};
+  const nvinfer1::Weights w{nvinfer1::DataType::kINT32, tmp, rank};
+  auto t = ctx->net->addConstant(d, w)->getOutput(0);
+  slice->setInput(idx, *t);
+}
+
+nvinfer1::ITensor* vec2Tensor(int32_t* dim, int rank, ConversionCtx* ctx) {

```

Comment:
Can we generalize this function and put it in utils or build it out of already existing utils?

Reply:
Thanks for you advice. I have moved this function to utils.